### PR TITLE
adding EFS_ENCRYPTED_CHECK rule

### PR DESCRIPTION
**Pull request by rafihm**
I confirm these files are made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)

*Issue #, if available:*
#129 
*Description of changes:*

### CODE REVIEW

File: python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py [hunk 1]
```diff
@@ -0,0 +1,448 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (jongogogo) [inline python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py]: Not the latest RDK template

File: python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+            break
+
+    return all_file_systems
+
+
+def evaluate_parameters(rule_parameters):
+
+    # If parameter is given check whether it's ARN, else ignore it as it is optional paramter.
+    if 'KmsKeyId' not in rule_parameters:
+        return False
```
Comment (jongogogo) [inline python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py]: I prefer to keep the output of functions constant. Always return a dict in this case.

Start with 
```python
if 'KmsKeyId' not in rule_parameters:
        return {}

File: python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+    if rule_parameters['KmsKeyId'] and 'arn:aws:kms' not in rule_parameters['KmsKeyId']:
```
Comment (jongogogo) [inline python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py]: The `rule_parameters['KmsKeyId']` is always True due to your guard clause above. You can remove it

File: python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+        raise ValueError('Invalid value for paramter KmsKeyId, Expected KMS Key ARN')
+    if 'KmsKeyId' in rule_parameters and "arn:aws:kms" in rule_parameters['KmsKeyId']:
```
Comment (jongogogo) [inline python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py]: `'KmsKeyId' in rule_parameters` is always True
`"arn:aws:kms" in rule_parameters['KmsKeyId']` is always True

You can just `return rule_parameters` without a `if` statement

File: python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK_test.py [hunk 5]
```diff
@@ -0,0 +1,253 @@
+import sys
```
Comment (jongogogo) [inline python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK_test.py]: Not the latest RDK template

File: python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK_test.py [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+rule = __import__('EFS_ENCRYPTED_CHECK')
+
+
+class ComplianceTestScenarios(unittest.TestCase):
+
+    rule_valid_parameters = '{"KmsKeyId": "arn:aws:kms:us-west-2:123456789012:key/12345-1111-1111-1111-123456789"}'
+    rule_invalid_value_parameter = '{"KmsKeyId": "asdfa97asf8a0sf09sa8df0a98sd0f8as0f8d0"}'
+    rule_empty_parameter = '{"KmsKeyId": ""}'
+
+    invoking_event_iam_role_sample = '{"configurationItem":{"relatedEvents":[],"relationships":[],"configuration":{},"tags":{},"configurationItemCaptureTime":"2018-07-02T03:37:52.418Z","awsAccountId":"123456789012","configurationItemStatus":"ResourceDiscovered","resourceType":"AWS::IAM::Role","resourceId":"some-resource-id","resourceName":"some-resource-name","ARN":"some-arn"},"notificationCreationTime":"2018-07-02T23:05:34.445Z","messageType":"ConfigurationItemChangeNotification"}'
```
Comment (jongogogo) [inline python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK_test.py]: Delete what is not necessary.

**Comment by rafihm** [comment]
Change Log 1:
 - Changed the template to latest version
 - Modified gherkin parameter section to remove the table format
 - Modified gherkin scenario 2
 - returning None if no file system exists
 - correcting type in annotation
 - removed maxItems from API call
 - removed extra variables present in the test file

Let me know if any modification requires. Thanks

File: python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+    evaluations = []
+
+    # check whether atlease one file system exist
+    if not all_file_systems:
+        return None
+
+    for each_efs in all_file_systems:
+        # check if file system is encrypted
+        if not each_efs['Encrypted']:
+            evaluations.append(build_evaluation(each_efs['FileSystemId'], 'NON_COMPLIANT', event, annotation='EFS is not encryprted'))
```
Comment (jongogogo) [inline python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py]: Typo in the annotation. Please call the services by the full name. Add a final dot.

File: python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+        # if there is no parameter, return COMPLIANT
+        if not valid_rule_parameters:
+            evaluations.append(build_evaluation(each_efs['FileSystemId'], 'COMPLIANT', event))
+            continue
+
+        # if valid parameter is provided then compare parameter with KmsKeyId
+        if each_efs['KmsKeyId'] == valid_rule_parameters:
+            evaluations.append(build_evaluation(each_efs['FileSystemId'], 'COMPLIANT', event))
+        else:
+            evaluations.append(build_evaluation(each_efs['FileSystemId'], 'NON_COMPLIANT', event, annotation='KMS key used to encrypt the EFS is not mathcing the KMS Key provided in the parameter'))
```
Comment (jongogogo) [inline python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py]: Same, find a better wordings.

**Comment by rafihm** [comment]
Modified Rule parameter and annotations wordings.

**Pull request merged**